pyemma/msm/tests/test_msm.py

```diff
@@ -1004,5 +1004,51 @@ def test_two_state_kinetics(self):
         self._two_state_kinetics(self.msm_sparse)
 
 
+class TestMSMMinCountConnectivity(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        dtraj = np.zeros(100, dtype=int)
+        dtraj[np.random.randint(0, 99, size=40)] = 1
+        dtraj[np.random.randint(0, 99, size=5)] = 2
+        dtraj[np.random.randint(0, 99, size=20)] = 3
```

Comment:
At his point there is no guarantee that you will keep state 2 in the unrestricted case. As you are drawing the time points independently of another, you can overwrite all '2s' with your '3s', which means you will loose state 2, and then the test for unrestricted connectivity will fail.